operator/metrics/legacy.go

```diff
@@ -0,0 +1,212 @@
+	"github.com/cilium/cilium/pkg/metrics/metric"
+)
+
+// Registry is the global prometheus registry for cilium-operator metrics.
+var Registry RegisterGatherer
+
+type RegisterGatherer interface {
+	prometheus.Registerer
+	prometheus.Gatherer
+}
+
+const (
+	// LabelStatus marks the status of a resource or completed task
+	LabelStatus = "status"
+
+	// LabelOutcome indicates whether the outcome of the operation was successful or not
+	LabelOutcome = "outcome"
+
+	// LabelOpcode indicates the kind of CES metric, could be CEP insert or remove
+	LabelOpcode = "opcode"
+
+	// Label values
+
+	// LabelValueOutcomeSuccess is used as a successful outcome of an operation
+	LabelValueOutcomeSuccess = "success"
+
+	// LabelValueOutcomeFail is used as an unsuccessful outcome of an operation
+	LabelValueOutcomeFail = "fail"
+
+	// LabelValueOutcomeAlive is used as outcome of alive identity entries
+	LabelValueOutcomeAlive = "alive"
+
+	// LabelValueOutcomeDeleted is used as outcome of deleted identity entries
+	LabelValueOutcomeDeleted = "deleted"
+
+	// LabelValueCEPInsert is used to indicate the number of CEPs inserted in a CES
+	LabelValueCEPInsert = "cepinserted"
+
+	// LabelValueCEPRemove is used to indicate the number of CEPs removed from a CES
+	LabelValueCEPRemove = "cepremoved"
+)
+
+var (
+	// IdentityGCSize records the identity GC results
+	IdentityGCSize = metrics.NoOpGaugeVec
+
+	// IdentityGCRuns records how many times identity GC has run
+	IdentityGCRuns = metrics.NoOpGaugeVec
+
+	// EndpointGCObjects records the number of times endpoint objects have been
+	// garbage-collected.
+	EndpointGCObjects = metrics.NoOpCounterVec
+
+	// CiliumEndpointSliceDensity indicates the number of CEPs batched in a CES and it used to
+	// collect the number of CEPs in CES at various buckets.
+	CiliumEndpointSliceDensity = metrics.NoOpHistogram
+
+	// CiliumEndpointsChangeCount indicates the total number of CEPs changed for every CES request sent to k8s-apiserver.
+	// This metric is used to collect number of CEP changes happening at various buckets.
+	CiliumEndpointsChangeCount = metrics.NoOpObserverVec
+
+	// CiliumEndpointSliceSyncTotal indicates the total number of completed CES syncs with k8s-apiserver by success/fail outcome.
+	CiliumEndpointSliceSyncTotal = metrics.NoOpCounterVec
+
+	// CiliumEndpointSliceSyncErrors used to track the total number of errors occurred during syncing CES with k8s-apiserver.
+	// This metric is going to be deprecated in Cilium 1.14 and removed in 1.15.
+	// It is replaced by CiliumEndpointSliceSyncTotal metric.
+	CiliumEndpointSliceSyncErrors = metrics.NoOpCounter
+
+	// CiliumEndpointSliceQueueDelay measures the time spent by CES's in the workqueue. This measures time difference between
+	// CES insert in the workqueue and removal from workqueue.
+	CiliumEndpointSliceQueueDelay = metrics.NoOpHistogram
+)
+
+type legacyMetrics struct {
+	IdentityGCSize                metric.Vec[metric.Gauge]
```

Comment:
Let's do a follow-up to remove `legacyMetrics`. Should be pretty doable to do all these after the CES PR lands?

Reply:
Definitely. Both the identity GC and the CES controller have been modularized, so we can switch to `cell.Metric` and remove all of this. :+1: